src/databases/OpenFOAM/visit_vtkOpenFOAMReader.cxx

```diff
@@ -59,7 +59,7 @@
 #include <vector>
 #include <vtksys/SystemTools.hxx>
 #include <sstream>
-#include <vtk_zlib.h>
+#include <zlib.h>
```

Comment:
Shouldn't this be conditioned on `HAVE_ZLIB_H`?

Reply:
I believe this reader should be conditioned on having zlib. I think it would be too difficult to wrap the code sections that deal with compressed files.

Reply:
When I look at at [visit_vtkOpenFoamReader.cxx](https://github.com/visit-dav/visit/blob/684e82f2e5a18f4f9495d49c3c066c70c8cf6059/src/databases/OpenFOAM/visit_vtkOpenFOAMReader.cxx#L1447), I see logic suggestive that it is capable of reading either compressed or UNcompressed files. On that simple check, my guess is that it should not get disabled with ZLIB is not present and instead, `#Iinclude <zlib.h>` should be conditined on `#HAVE_ZLIB_H`

Reply:
As I stated before, I think the logic in this file is too complex to wrap all the gz stuff with preprocessor checks.

Reply:
Ok, so I guess I was misunderstanding...you are saying that it is NOT a simple matter of including or not including `zlib.h` to support compressed inputs here because there are also many other places that the file would need conditional compilation blocks